src/MongoDB.Bson/IO/JsonWriter.cs

```diff
@@ -204,6 +209,23 @@ public override void WriteDateTime(long value)
                     _textWriter.Write("{{ \"$date\" : {0} }}", value);
                     break;
 
+                case JsonOutputMode.RelaxedExtendedJson:
+                    if (value >= BsonConstants.DateTimeMillisecondsStartEpoch &&
+                        value <= BsonConstants.DateTimeMaxValueMillisecondsSinceEpoch)
+                    {
+                        var utcDateTime = BsonUtils.ToDateTimeFromMillisecondsSinceEpoch(value);
+                        _textWriter.Write("{{ \"$date\" : \"{0}\" }}", utcDateTime.ToString("yyyy-MM-ddTHH:mm:ss.FFFZ"));
+                    }
+                    else
+                    {
+                        _textWriter.Write("{{ \"$date\" : {{ \"$numberLong\" : \"{0}\" }} }}", value);
```

Comment:
Investigate whether this should be:

     _textWriter.Write("{{ \"$date\" : {{ \"$numberLong\" : \"{0}\" }} }}", value.ToString(NumberFormatInfo.InvariantInfo));

We sometimes have issues with culture dependent ToString conversions.

It's possible that integer types convert to string the same way in all cultures, although I am suspicious that negative number might format differently in different cultures.

If you find that converting integers to string IS culture dependent, there are multiple other places in this file that might need to to refactored.

Additionally, perhaps we should instead be calling :

    JsonConvert.ToString(value);

The JsonConvert class is intended to centralize all conversions of values to and from JSON string representations.

Reply:
A quick repro I made:
```
long number = -1;
var cultures = CultureInfo.GetCultures(CultureTypes.AllCultures);
foreach (var culture in cultures)
{
	var cultureSpecificRepresentation = number.ToString(culture);
	if (cultureSpecificRepresentation != number.ToString())
	{
		Console.WriteLine(cultureSpecificRepresentation);
	}
}
```
The minus sign does not differ across all 848 cultures.
In fact one can change the minus sign value programatically (https://stackoverflow.com/a/8492527/6834435), but I think we shouldn't account for this case.
Not done.